src/fpmath.h

```diff
@@ -47,7 +47,7 @@
 #define _PDP_ENDIAN       __ORDER_PDP_ENDIAN__
 #define _BYTE_ORDER       __BYTE_ORDER__
 
-#elif defined(__linux)
+#if defined(__linux__) || defined(__FreeBSD_kernel__) || defined(__GNU__)
```

Comment:
Debian/kFreeBSD has /usr/include/features.h, /usr/include/endian.h *and* /usr/include/machine/endian.h.

I tried building both ways, but including <machine/endian.h> resulted in the library having an executable stack.